packages/nimble-components/src/banner/specs/README.md

```diff
@@ -0,0 +1,167 @@
+# Banner
+
+## Overview
+
+The banner is a component used to display a persistent notification to a user.
+
+### Background
+
+[Design specification](https://xd.adobe.com/view/33ffad4a-eb2c-4241-b8c5-ebfff1faf6f6-66ac/screen/29c405f7-08ea-48b6-973f-546970b9dbab/)
+
+[GitHub issue](https://github.com/ni/nimble/issues/305)
+
+### Non-goals
+
+- Arbitrary HTML content
+- A container component that would enforce design constraints on multiple banners displayed together
+- Limitating length of displayed text and/or truncating text with ellipsis
+
+### Features
+
+- Optionally dismissable
+- Optional link or button (only 1) to perform an action/navigation
+- Distinct designs for banners of type Error, Warning, Info, or Neutral
+
+### Risks and Challenges
+
+None
+
+### Prior Art/Examples
+
+Refer to examples in the "Some examples of screenshots" section of the [GitHub issue](https://github.com/ni/nimble/issues/305).
+
+---
+
+## Design
+
+The banner is a standalone component that spans the width of its containing element, presenting, from left to right, an icon (dependent on Error/Warning/Info type), an optional title, an optional plaintext message, an optional action button or link, and an optional close control. The banner is intended to display short messages (recommended three lines or fewer) that communicate some persistent state or condition.
+
+### API
+
+*The key elements of the component's public API surface:*
+
+- *Component Name*: `nimble-banner`
+- *Props/Attrs*:
+  - `heading` - if set, a heading with the specified text is shown to the left of the message. OPEN QUESTION: should we instead use the existing `title` attribute?
+  - `text` - the message text
+  - `prevent-dismiss` - set to hide the close button (attr name taken from Nimble dialog)
+  - `type` - one of `error`, `warning`, `info`, or `neutral` (default)
+  - `action-text` - the text of the action link/button. If unset, no link/button is displayed.
+  - `action-href` - if set, the action element will be a link with the given `href`. (Otherwise, when `action-text` is set, a button element will be displayed.)
+  - `action-button-appearance` - either `outline` or `ghost` (default). When an action button is displayed, this controls the appearance variant.
+- *Methods*
+- *Events*
+  - `close` - fired when the banner is dismissed
```

Comment:
When the user activates the dismiss button, does the button contain code to dismiss itself or does it only fire the event? Assuming it's the latter, it would be good to clarify with something like:
```suggestion
  - `close` - fired when the banner's dismiss button is pressed. The client is responsible for removing the banner when this happens.
```

Reply:
I assumed we would hide the banner ourselves. I can't think of any reason to leave it up to the client.

Reply:
I guess the challenge is in deciding how to hide it. Remove from the DOM vs ‘display: none’ vs ‘visibility: hidden’? We could pick one that works for most clients but possibly some clients would want a different approach. And what we pick might actually inhibit some client options, like if they wanted to animate it away but we already hid it. Leaving it entirely up to clients gives more flexibility, at the cost of some ease-of-use.

Reply:
Wouldn't animation vs no animation be one of the decisions the design system should make for the client? Maybe that's a fuzzy line. As far as `display:none` vs `visibility:hidden`, I think the clear choice in our case is the former. We may be stacking multiple banners, and we don't want to reserve space for any/all possible banners.

Ultimately, I don't feel strongly either way, but I'm leaning slightly toward doing the hiding ourselves.

Reply:
>Wouldn't animation vs no animation be one of the decisions the design system should make for the client? 
I agree it's hard to know. The animation *might* be context-sensitive (e.g. appear from the top if the banner's at the top of the screen but from the side if it's at the side of the screen or not animate at all if it's visible when the page loads). If we provide it from Nimble I think we'd at least need to make it configurable.

I'm coming around to Nimble being responsible for hiding it (agree that `display:none` is the right approach), assuming we address Molly's points about re-showing it and striving for API consistency with other dismissable components.

Reply:
Comparing to native apis, I think this element aligns more closely to the [details](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details) element which provides an expand / collapse container.

That api has an open attribute and uses a toggle event (fires after close). That would more closely match the api surface we are adding to [menu button](https://github.com/ni/nimble/pull/986) as well where we pulled that from the [popover spec](https://github.com/whatwg/html/pull/8717#issuecomment-1386178070).

So api-wise it kinda is a popover, a visual thing that comes up and goes away. So I think aligning on these api patterns makes sense. It's also not like `dialog` or `drawer` where the extra energy on tracking exactly when it opens and closes is as useful, so I don't think it warrants the same promise-based api machinations we did there yet.

boolean attribute `open` 
and events `toggle` with details newState and oldState.

(for consistency with the menu button we could also implement `beforetoggle` now as well, but I don't see that as being super useful because "opening" a banner is only programmatic so the client always controls it unlike a menu button which opens from user interaction).

Can style the banner as display: none; based on the open attribute and clients can override in css based on attribute, for example visibility: hidden; if they want the banner to preserve space instead of not preserve space when hidden.